script.sh

```diff
@@ -162,9 +162,15 @@ parse_defs_C()
     tmp=`mktemp -d`
     full_path=$tmp/$opt2
     git cat-file blob "$opt1" > "$full_path"
+
+    # Works for, e.g., .c and .h
     ctags -x --kinds-c=+p+x "$full_path" |
     grep -avE "^operator |CONFIG_" |
     awk '{print $1" "$2" "$3}'
+
+    # Works for, e.g., .S
+    perl -ne '/^\s*ENTRY\((.+?)\)/ and print "$1 function $.\n"' "$full_path"
```

Comment:
At first I was thinking about making sure that it only applies to .S files but it seems to be also used in C files and because this is only called for C/CPP/ASM files we are sure that Documentation files will not be indexed :sweat_smile: 

I listed all the occurences of ENTRY in linux and maybe we will have problems for : 
- [arch/arm/lib/bitops.h](https://elixir.bootlin.com/linux/latest/source/arch/arm/lib/bitops.h#L7)
- [arch/arm/xen/hypercall.S](https://elixir.bootlin.com/linux/latest/source/arch/arm/xen/hypercall.S#L58)
- [arch/arm/mm/proc-macros.S](https://elixir.bootlin.com/linux/latest/source/arch/arm/mm/proc-macros.S#L286)

So maybe this will be better : 
```suggestion
    perl -ne '/^\s*ENTRY\((\w+)\)/ and print "$1 function $.\n"' "$full_path"
```
<br/>

And it looks like ENTRY can also be used like [that](https://elixir.bootlin.com/linux/latest/source/drivers/firmware/broadcom/bcm47xx_sprom.c#L194) ...

But maybe there are other macros we need to support, I don't know :thinking:
But it will always be possible to modify that later if we need to add something else :smile:

Reply:
Good catch - looks good to me!

Reply:
I made this change.  I also added some tests for non-`\w`, for a little added confidence.